include/mc_control/ControllerClient.h

```diff
@@ -588,14 +588,20 @@ struct MC_CONTROL_CLIENT_DLLAPI ControllerClient
    *
    * \p values Possible values
    *
+   * \p def default index
+   *
    * \p send_index If true, the implementation should send back the index
    * rather than the value
+   *
+   * \p default_from_user whether the default value is provided from user
    */
   virtual void form_combo_input(const ElementId & /*formId*/,
                                 const std::string & /*name*/,
                                 bool /*required*/,
                                 const std::vector<std::string> & /*values*/,
-                                bool /*send_index*/)
+                                int /*def*/,
+                                bool /*send_index*/,
+                                bool /*default_from_user*/)
```

Comment:
for backward compatibility we need to keep two virtual functions like in the case of `form_integer_input`, the first is the original virtual method with no implementation and the second is thew one calling the first. The client only need to implement the new one but clients that implement the old one will keep working